### PR TITLE
Bigint and sqrt changes

### PR DESCRIPTION
**Pull request by ebfull**
This brings in @daira's changes from #5 and also ensures that Fp/Fp2 sqrt returns zero.

### CODE REVIEW

**Comment by ebfull** [comment]
I also just threw in a patch that removes the debugging output that libsnark was producing when you construct proofs.

```
* Elements of w skipped: 0 (0.00%)
* Elements of w processed with special addition: 0 (0.00%)
* Elements of w remaining: 247 (100.00%)
* Elements of w skipped: 0 (0.00%)
* Elements of w processed with special addition: 0 (0.00%)
* Elements of w remaining: 251 (100.00%)
* Elements of w skipped: 0 (0.00%)
* Elements of w processed with special addition: 0 (0.00%)
* Elements of w remaining: 250 (100.00%)
* Elements of w skipped: 0 (0.00%)
* Elements of w processed with special addition: 0 (0.00%)
* Elements of w remaining: 252 (100.00%)
* G1 elements in proof: 7
* G2 elements in proof: 1
* Proof size in bits: 2294
```

... since it throws that into stdout.

File: src/algebra/fields/fp2.tcc [hunk 1]
```diff
@@ -145,6 +145,10 @@ Fp2_model<n,modulus> Fp2_model<n,modulus>::Frobenius_map(unsigned long power) co
 template<mp_size_t n, const bigint<n>& modulus>
 Fp2_model<n,modulus> Fp2_model<n,modulus>::sqrt() const
 {
+    if (is_zero()) {
+        return *this;
+    }
+
```
Comment (daira) [inline src/algebra/fields/fp2.tcc:151]: Fp3 has the same problem (for consistency, we should also remove the `#if DEBUG` around the Euler check on that one). I don't think we ever call sqrt on an Fp3 instance, but it's ugly for it to be inconsistent.

**Comment by daira** [comment]
utACK on the commits you've added, modulo the comment about Fp3.

Tests for sqrt would be nice.

**Comment by daira** [comment]
It looks like there's a test for sqrt in test_fields.cpp, so it could be added there (but note that https://github.com/zcash/libsnark/pull/6/commits/213e2881743f59be6b6f5323f2ffbd872c13b100 disabled building the executable for that since I couldn't get it to compile, due to removed code).

**Comment by daira** [comment]
I filed https://github.com/zcash/zcash/issues/1544 to fix Fp3. Post-hoc ACK.

**Pull request merged**